### PR TITLE
Set *.png and *.ico to be treated as binary files.

### PR DESCRIPTION
**Pull request by Nicholas42**
The change b732beba6 makes all files text files and sets lf as eol. This will automatically change all files to have lf if they are touched by git (e.g. via git status). This change cannot be stashed and makes it hard to work with the repo (rebase and checkout don't really work). See also #6183.

Here, we set the offending files (*.png and *.ico, but that might be more in the future) to be treated as binary files and not be changed by git.

### CODE REVIEW

**Comment by FellowTraveler** [comment]
**The repo will be unusable with branches until this is merged.** This is the most critical PR rn IMO.

**Comment by PAN-Chuwen** [comment]
Might also need to include `*.icns` in the binary, as the following file is listed as modified: `macapp/assets/icon.icns`.

**Comment by Nicholas42** [comment]
> Might also need to include `*.icns` in the binary, as the following file is listed as modified: `macapp/assets/icon.icns`.

Interesting, it wasn't flagged for me. But I added it now.

**Comment by Nicholas42** [comment]
> **The repo will be unusable with branches until this is merged.** This is the most critical PR rn IMO.

@FellowTraveler You might be able to temporarily override that behavior until this is merged by overriding the git attributes locally: https://stackoverflow.com/a/50950870/12345656

**Comment by mxyng** [comment]
The reason `.gitattributes` was added was to resolve an issue with the go linter. Instead, I think we can partially revert the previous change to `.gitattributes` and explicitly mark `*.go` as LF.

e.g.

```
* text=auto
*.go text eol=lf
```

I'm happy to merge this change if you make this adjustment

**Comment by sammcj** [comment]
Ah - glad to see someone has a solution to this, it caused me [headaches](https://github.com/ollama/ollama/pull/5894#issuecomment-2277316957) when trying to update a PR yesterday, I ended up doing all sorts of hacks to work around it which eventually sent Github crazy and I had to bin the PR and create a new one!

**Review by mxyng — APPROVED** [review]
Thanks for fixing

**Pull request merged**
